### PR TITLE
app: buildkite pipeline for releases

### PR DESCRIPTION
**Pull request by burmudar**
This is mostly constrained to the `enterprise/dev/app` directory.

- [x] undo the bazel changes

- Builds the linux version on Bazel agents
- Builds arm64 version on macos
- Builds x86_64 version on macos using go - There is a fix to the bazel scripts I have to make for this
- Bundles the linux versions on linux using Tauri
- Bundles the macos version on MacOS + Signs all the artefacts with Apple
- Creates a GitHub release

## Future work
Currently a x86 on MacOS isn't published / bundled since we have to install specific rust libraries for that

## Test plan
https://buildkite.com/sourcegraph/sourcegraph-app-release/builds?branch=wb%2Fapp%2Faws-macos

### CODE REVIEW

**Review by davejrt — APPROVED** [review]

File: .aspect/bazelrc/ci.macos.bazelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# From https://github.com/aspect-build/aspect-cli-plugin-template/blob/main/.github/workflows/ci.bazelrc
+build --announce_rc
```
Comment (jhchabran) [inline .aspect/bazelrc/ci.macos.bazelrc:2]: we already have this in another bazelrc :)

File: enterprise/dev/app/build-backend.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    -o .bin/sourcegraph-backend-${platform} \
+    -trimpath \
+    -tags dist \
+    -ldflags "$ldflags" \
+    ./enterprise/cmd/sourcegraph
+}
+
+bazelrc() {
+  if [[ $(uname -s) == "Darwin" ]]; then
+    echo "--bazelrc=.bazelrc --bazelrc=.aspect/bazelrc/ci.macos.bazelrc"
```
Comment (jhchabran) [inline enterprise/dev/app/build-backend.sh:42]: We should include include `ci.bazelrc` which gets the defaults from Aspect (including the announce_rc thing for example)

File: enterprise/dev/app/detect_platform.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  local platform=""
+  case "$(uname -m)" in
+    "amd64")
+      arch="x86_64"
+      ;;
+    "arm64")
+      arch="aarch64"
+      ;;
+    *)
+      arch=$(uname -m)
```
Comment (jhchabran) [inline enterprise/dev/app/detect_platform.sh:16]: Do we need that catch all? Or should this be an error?

**Review by jhchabran — CHANGES_REQUESTED** [review]
Nothing really blocking, but I'd like to know why we're not sourcing some of the bazelrcs :)

File: enterprise/dev/app/build-backend.sh (hunk 2 above)
Comment (burmudar) [reply enterprise/dev/app/build-backend.sh:42]: Fixing!

File: enterprise/dev/app/detect_platform.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  case "$(uname -s)" in
+    "Darwin")
+      platform="${arch}-apple-darwin"
+      ;;
+    "Linux")
+      platform="${arch}-unknown-linux-gnu"
+      ;;
+    *)
+      platform="${arch}-unknown-unknown"
```
[the remark this answers is not in the document]
Comment (burmudar) [reply enterprise/dev/app/detect_platform.sh:27]: Will remove!

File: enterprise/dev/app/detect_platform.sh (hunk 3 above)
Comment (burmudar) [reply enterprise/dev/app/detect_platform.sh]: fixed!

File: .aspect/bazelrc/ci.macos.bazelrc (hunk 1 above)
Comment (burmudar) [reply .aspect/bazelrc/ci.macos.bazelrc]: updated and removed

**Review by slimsag — APPROVED** [review]
LGTM, I am gonna push this through / merge this now so we can get to some demos :)

**Review by jhchabran — APPROVED** [review]

**Comment by jhchabran** [comment]
Approved so you can fix the shell linting issues on your own time :)

**Pull request merged**